### PR TITLE
A few easy fixes sprinkled all around

### PR DESCRIPTION
**Pull request by evertvorster**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [x] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only

Issues:
- Please list any related issues
#230 - Unused variable 'mis2' - Partial fix, there are more problems there
#232 `*` in boolean context
#233 - if clause does not protect - Partial fix, there are more problems there
#234 - unused variable 'div2pi'
#236 - unnecessary parenthesis in declaration 'g' 
#237 - unused variable 'fightername'
#238 - variable 'color' set and not used
#242 - variable 'iter' set and not used
#243 - `*` in boolean context
#244 - unused variable 'itemscale'
#247 - unused variable 'network_draw_messages'

Purpose:
- What is this pull request trying to do?
Clean up the compile logs. There are several warnings there.
- What release is this for?
Any
- Is there a project or milestone we should apply this to?
Not really

### CODE REVIEW

File: engine/src/cmd/damageable.cpp [hunk 1]
```diff
@@ -351,7 +351,7 @@ float Damageable::totalShieldEnergyCapacitance( const Shield &shield )
         XMLSupport::parse_float( vs_config->getVariable( "physics", "shield_energy_capacitance", ".2" ) );
     static bool  use_max_shield_value =
         XMLSupport::parse_bool( vs_config->getVariable( "physics", "use_max_shield_energy_usage", "false" ) );
-    return shieldenergycap*use_max_shield_value ? totalShieldVal( shield ) : currentTotalShieldVal( shield );
+    return shieldenergycap*(use_max_shield_value ? totalShieldVal( shield ) : currentTotalShieldVal( shield ));
```
Comment (stephengtuggy) [inline engine/src/cmd/damageable.cpp:354]: This looks correct

File: engine/src/cmd/drawable.cpp [hunk 2]
```diff
@@ -29,43 +29,43 @@ bool Drawable::DrawableInit(const char *filename, int faction,
...
+                    meshes->push_back( m );
+    #ifdef DEBUG_MESH_ANI
+        // TODO: search and replace cerr with BOOST_LOG_TRIVIAL(error)
+                    cerr << "Animated Mesh: " << path << " loaded - with: " << m->getVertexList()->GetNumVertices() << " vertices." << endl;
+    #endif
+            }
+            else
+                    break;
+            i++;
+    }
```
Comment (stephengtuggy) [inline engine/src/cmd/drawable.cpp:68]: OK, this does look like the correct fix after all.

File: engine/src/cmd/drawable.cpp [hunk 3]
```diff
@@ -120,8 +120,8 @@ void Drawable::AnimationStep()
         if(nextactiveMesh >= vecAnimations.at(activeAnimation)->size())
             nextactiveMesh = 0;
 
-            if(loopCount > 0)
-                loopCount--;
+        if(loopCount > 0)
+            loopCount--;
```
Comment (stephengtuggy) [inline engine/src/cmd/drawable.cpp:124]: This looks like the correct fix also.

File: engine/src/cmd/planetary_orbit.cpp [hunk 4]
```diff
@@ -37,8 +37,7 @@ PlanetaryOrbit::PlanetaryOrbit( Unit *p,
         subtype = (SLOCATION);
         AttachOrder( centre );
     }
-    const double div2pi = ( 1.0/(2.0*PI) );
-
```
Comment (stephengtuggy) [inline engine/src/cmd/planetary_orbit.cpp:41]: OK, cool.

File: engine/src/cmd/unit_generic.cpp [hunk 5]
```diff
@@ -1312,7 +1312,8 @@ void Unit::Fire( unsigned int weapon_type_bitmask, bool listen_to_owner )
         return;
     }
     unsigned int mountssize = mounts.size();
-    int playernum = _Universe->whichPlayerStarship( this );
+//    int playernum = 
+    _Universe->whichPlayerStarship( this );
```
Comment (stephengtuggy) [inline engine/src/cmd/unit_generic.cpp:1316]: Yep, this should be fine. I believe this was only used by the multiplayer game.

File: engine/src/cmd/unit_generic.cpp [hunk 6]
```diff
@@ -1398,15 +1399,6 @@ void Unit::Fire( unsigned int weapon_type_bitmask, bool listen_to_owner )
...
     }
-    if ( !gunFireRequests.empty() ) {
-
-            char mis2 = false;
-    }
-
-    //Client missile requests can be grouped because clients only send a boolean, not a serial.
-    if ( !missileFireRequests.empty() ) {
-        char mis2 = true;
-    }
```
Comment (stephengtuggy) [inline engine/src/cmd/unit_generic.cpp:1409]: This looks fine

File: engine/src/cmd/unit_generic.cpp [hunk 7]
```diff
@@ -6314,12 +6306,11 @@ void Unit::EjectCargo( unsigned int index )
 
     Cockpit *cp = NULL;
     if ( index == (UINT_MAX-1) ) {
-        int pilotnum = _Universe->CurrentCockpit();
+        _Universe->CurrentCockpit();
```
Comment (stephengtuggy) [inline engine/src/cmd/unit_generic.cpp:6309]: You can probably remove the `_Universe->CurrentCockpit();` part as well

File: engine/src/cmd/unit_generic.cpp [hunk 8]
```diff
@@ -7865,14 +7856,11 @@ float Unit::DealDamageToHull( const Vector &pnt, float damage)
     }
 
   // Play Damage Sound
-  static float damage_factor_for_sound =
       XMLSupport::parse_float( vs_config->getVariable( "audio", "damage_factor_for_sound", ".001" ) );
```
Comment (stephengtuggy) [inline engine/src/cmd/unit_generic.cpp:7859]: You can probably remove the `XMLSupport::...` part as well

File: engine/src/gfx/nav/navscreen.cpp [hunk 9]
```diff
@@ -1769,7 +1769,7 @@ void NavigationSystem::TranslateCoordinates( QVector &pos,
     // as well as to display system names and make systems
     // highlight their names on mouse hover
     // and also make them clickable for setting jump routes
-    float itemscale = CalculatePerspectiveAdjustment(
+    CalculatePerspectiveAdjustment(
```
Comment (stephengtuggy) [inline engine/src/gfx/nav/navscreen.cpp:1772]: I **think** this is correct. Call function/method CalculatePerspectiveAdjustment, then throw away the result.

File: engine/src/gfx/vdu.cpp [hunk 10]
```diff
@@ -841,7 +841,6 @@ void VDU::DrawTarget( GameCockpit *cp, Unit *parent, Unit *target )
 
 void VDU::DrawMessages( GameCockpit *parentcp, Unit *target )
 {
-    static bool network_draw_messages = XMLSupport::parse_bool( vs_config->getVariable( "graphics", "network_chat_text", "true" ) );
```
Comment (stephengtuggy) [inline engine/src/gfx/vdu.cpp:844]: I think this is fine, at least for now

File: engine/src/gfx/vdu.cpp [hunk 11]
```diff
@@ -1738,9 +1737,6 @@ bool VDU::SetWebcamAnimation()
 void VDU::DrawWebcam( Unit *parent )
 {
     using VSFileSystem::JPEGBuffer;
-    int   length;
-    char *netcam;
-
```
Comment (stephengtuggy) [inline engine/src/gfx/vdu.cpp:1743]: Should be fine

File: engine/src/gfx/nav/navscreen.cpp (hunk 9 above)
Comment (evertvorster) [reply engine/src/gfx/nav/navscreen.cpp:1772]: Here I am just going by a conversation we had on gitter. Hopefully I will understand more about what I am doing in the future. Thanks for double-checking!

**Pull request merged**